Lib/site.py

```diff
@@ -448,6 +448,14 @@ def write_history():
                     # home directory does not exist or is not writable
                     # https://bugs.python.org/issue19891
                     pass
+                except OSError:
+                    if not os.access(history, os.W_OK):
+                        print("Permission error!, unable to write .python_history")
```

Comment:
I don't think it should separately handle a permission error that's not categorized correctly due to a GNU History bug, which is incorrectly returning -1 as an errno value. I analyzed this in detail on the tracker issue bpo-39468. (Note that an errno value of -1 must be a bug because C defines `errno` as a macro that "expands to a modifiable lvalue that has type int and ... is set to a *positive* error number by several library functions".) If this bug gets fixed upstream, then a proper `PermissionError` will be raised for this case. 

If any message is written to `sys.stderr` (not `sys.stdout`) when updating the history file fails, then I think it should be a generic notification, rather than speculating on the reason. The user can figure out what's wrong. A precedent shouldn't be set for a never-ending list of platform-specific failure cases.

Reply:
The solution was simpler than I thought.
Just catching OSError exception resolves the issue for now, until it gets resolved in readline itself.